### ISSUE TITLE
Grave Pet not restoring my items

### ISSUE THREAD

**Issue by InkerJ** [capnkirok/Inventory-Pets#416]
I've been testing it because I haven't been able to figure it out. But I haven't found out how it even works. I even lost all of my stuff in a server, though I had a grave pet.

**Comment by capnkirok** [comment]
Do you have any other mods installed that restore items? Can you post your modlist? Also what version of minecraft and inventory pets are you using?

**Comment by InkerJ** [comment]
I have Biome's O Plenty, Just Enough Items, Additional Structures, AtmoicStryker's Battle Towers, Better Animal Models, Better Foliage, Biblio Craft, ElecCore, Huds - Armor Status & Damage Indicators, Ice and Fire: Dragons, Inventory Pets, LLibrary, Roguelike Dungeons, Shadowfact's Forgelin, So Many Enchantments, and The Twighlight Forest. I'm using the 1.12.2-2.0.12 version

**Comment by capnkirok** [comment]
Thanks, I'll check it out.

**Comment by ramonGonzEdu** [comment]
I had the same issue on my 1.16.5 server where the grave pet worked 50% of the time (we had like 4 bones in the inventory)

**Comment by capnkirok** [comment]
@ramonGonzEdu can you share the other mods you have installed on your server?

**Comment by RGFTheCoder** [comment]
```js
tetra-1.16.5-3.8.0
cc-tweaked-1.16.4-1.95.3
create-mc1.16.3_v0.3e
fairylights-4.0.5-1.16.5
inventorypets-2.0.12
thermal_foundation-1.16.3-1.1.6
mgui-1.16.4-3.1.3
cofh_core-1.16.3-1.1.6
jei-1.16.4-7.6.1.72
ftb-gui-library-1605.2.1.41-forge
ftb-ultimine-1605.2.0-build.25-forge
ftb-chunks-1605.2.3-build.71
ftb-teams-1604.1.0.16-forge
architectury-1.8.131-forge
cloth-config-4.11.15-forge
EnderStorage-1.16.4-2.7.1.166-universal
CodeChickenLib-1.16.4-3.5.1.408-universal
furniture-7.0.0-pre19-1.16.3
```

**Comment by capnkirok** [comment]
@ramonGonzEdu well, I just installed all of these mods and tried it out on a 1.16.5 Forge Server. I killed my self over 50 times, using a variety of different methods, and the Grave Pet restored all items every time.

Let me ask a few more questions:
- Are you hearing the sound of the Grave Pet when you die? Do you hear whats sounds like being buried under gravel?
- Are you keeping your items stored in another storage container like a backpack or one of the inventory pets storage containers?
- Just out of curiosity, does it matter how many people are on the server when this happens?

I'm going to continue to look at this.

**Comment by RGFTheCoder** [comment]
It seems to work consistently now, but I don't have sound; I believe it might've been that I ran out of bones and the icon looked similar, but I keep most pets in my hotbar, so I believe this is a code ID10T on my part

**Comment by capnkirok** [comment]
Thank you for the follow up. You are not the only one that this has happened to. It's hard to see the durability bar with this pet in particular. I may have to update this.

**Comment by capnkirok** [comment]
Updating this thread. I discovered a bug in my code that may have caused the Grave Pet not to work in a certain situation, namely, if the item you were trying to restore was in the hotbar and had more than 1 item damage (e.g., any pickaxe or sword that had been used at least once). This could have in fact caused the seemingly random behavior where the Grave Pet works sometimes but not others. At any rate, this has been fixed in the development environment and will be in the next release (2.0.13).

This only applies to the 1.15 and 1.16 versions.
